### ISSUE TITLE
Nested Mutation BelongsTo: Undefined property

### ISSUE THREAD

**Issue by theovier** [nuwave/lighthouse#1504]
**Describe the bug**
Executing a mutation to create a model with a belongsTo relation returns `"Undefined property: App\\Meal::$author"`.

**Expected behavior/Solution**

The model is created and connected with an existing model via its belongsTo relation.
I tried to closely follow the steps listed in the [tutorial about nested mutations](https://lighthouse-php.com/master/eloquent/nested-mutations.html#belongsto). However, I was unable to get the desired functionality working.

**Steps to reproduce**

Schema.graphql:
```
type User {
    id: ID!
    name: String!
}

type Meal {
    id: ID!
    name: String!
    author: User! @belongsTo
}

type Mutation {
    createMeal(input: CreateMealInput! @spread): Meal @create
}

input CreateMealInput {
    name: String!
    author: CreateAuthorRelation!
}

input CreateAuthorRelation {
    connect: ID
}
```

Meal.php
```
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Meal extends Model {
    protected $guarded = [];

    public function author(): BelongsTo {
        return $this->belongsTo(User::class, 'author');
    }
}
```

Calling the mutation:
```
mutation {
  createMeal(input: { name: "my meal", author: { connect: 1 } }) {
    id
    name
  }
}
```

Note that a user with the ID 1 exists, as queries against user(s) are successful. 

Meal Migration:
```
Schema::create('meals', function (Blueprint $table) {
            $table->id();
            $table->string('name');
            $table->foreignId('author')->references('id')->on('users');
        });
```

**Output/Logs**

<details><summary>Click to expand</summary>

```
{
  "errors": [
    {
      "debugMessage": "Undefined property: App\\Meal::$author",
      "message": "Internal server error",
      "extensions": {
        "category": "internal"
      },
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createMeal"
      ],
      "trace": [
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php",
          "line": 101,
          "call": "Illuminate\\Foundation\\Bootstrap\\HandleExceptions::handleError(8, 'Undefined property: App\\Meal::$author', '/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php', 101, array(1))"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/Relation.php",
          "line": 71,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\BelongsTo::addConstraints()"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php",
          "line": 71,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\Relation::__construct(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php",
          "line": 233,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\BelongsTo::__construct(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal, 'author', 'id', 'author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php",
          "line": 217,
          "call": "Illuminate\\Database\\Eloquent\\Model::newBelongsTo(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal, 'author', 'id', 'author')"
        },
        {
          "file": "/var/www/app/Meal.php",
          "line": 21,
          "call": "Illuminate\\Database\\Eloquent\\Model::belongsTo('App\\User', 'author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php",
          "line": 434,
          "call": "App\\Meal::author()"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php",
          "line": 420,
          "call": "Illuminate\\Database\\Eloquent\\Model::getRelationshipFromMethod('author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php",
          "line": 375,
          "call": "Illuminate\\Database\\Eloquent\\Model::getRelationValue('author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php",
          "line": 1633,
          "call": "Illuminate\\Database\\Eloquent\\Model::getAttribute('author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php",
          "line": 101,
          "call": "Illuminate\\Database\\Eloquent\\Model::__get('author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/Relation.php",
          "line": 71,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\BelongsTo::addConstraints()"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/BelongsTo.php",
          "line": 71,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\Relation::__construct(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php",
          "line": 233,
          "call": "Illuminate\\Database\\Eloquent\\Relations\\BelongsTo::__construct(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal, 'author', 'id', 'author')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php",
          "line": 217,
          "call": "Illuminate\\Database\\Eloquent\\Model::newBelongsTo(instance of Illuminate\\Database\\Eloquent\\Builder, instance of App\\Meal, 'author', 'id', 'author')"
        },
        {
          "file": "/var/www/app/Meal.php",
          "line": 21,
          "call": "Illuminate\\Database\\Eloquent\\Model::belongsTo('App\\User', 'author')"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Execution/Arguments/SaveModel.php",
          "line": 55,
          "call": "App\\Meal::author()"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Execution/Arguments/ResolveNested.php",
          "line": 36,
          "call": "Nuwave\\Lighthouse\\Execution\\Arguments\\SaveModel::__invoke(instance of App\\Meal, instance of Nuwave\\Lighthouse\\Execution\\Arguments\\ArgumentSet)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Schema/Directives/MutationExecutorDirective.php",
          "line": 103,
          "call": "Nuwave\\Lighthouse\\Execution\\Arguments\\ResolveNested::__invoke(instance of App\\Meal, instance of Nuwave\\Lighthouse\\Execution\\Arguments\\ArgumentSet)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Support/Utils.php",
          "line": 91,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\MutationExecutorDirective::Nuwave\\Lighthouse\\Schema\\Directives\\{closure}(instance of Nuwave\\Lighthouse\\Execution\\Arguments\\ArgumentSet)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Schema/Directives/MutationExecutorDirective.php",
          "line": 105,
          "call": "Nuwave\\Lighthouse\\Support\\Utils::applyEach(instance of Closure, instance of Nuwave\\Lighthouse\\Execution\\Arguments\\ArgumentSet)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Schema/Directives/MutationExecutorDirective.php",
          "line": 55,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\MutationExecutorDirective::executeMutation(instance of App\\Meal, instance of Nuwave\\Lighthouse\\Execution\\Arguments\\ArgumentSet)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php",
          "line": 28,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\MutationExecutorDirective::Nuwave\\Lighthouse\\Schema\\Directives\\{closure}(instance of Illuminate\\Database\\MySqlConnection)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Schema/Directives/MutationExecutorDirective.php",
          "line": 67,
          "call": "Illuminate\\Database\\Connection::transaction(instance of Closure)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Schema/Factories/FieldFactory.php",
          "line": 166,
          "call": "Nuwave\\Lighthouse\\Schema\\Directives\\MutationExecutorDirective::Nuwave\\Lighthouse\\Schema\\Directives\\{closure}(null, array(2), instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 632,
          "call": "Nuwave\\Lighthouse\\Schema\\Factories\\FieldFactory::Nuwave\\Lighthouse\\Schema\\Factories\\{closure}(null, array(1), instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 555,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveOrError(instance of GraphQL\\Type\\Definition\\FieldDefinition, instance of GraphQL\\Language\\AST\\FieldNode, instance of Closure, null, instance of Nuwave\\Lighthouse\\Schema\\Context, instance of GraphQL\\Type\\Definition\\ResolveInfo)"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 478,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveField(GraphQLType: Mutation, null, instance of ArrayObject(1), array(1))"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 921,
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}(array(0), 'createMeal')"
        },
        {
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}(array(0), 'createMeal')"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 923,
          "function": "array_reduce(array(1), instance of Closure, array(0))"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 494,
          "call": "GraphQL\\Executor\\ReferenceExecutor::promiseReduce(array(1), instance of Closure, array(0))"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 256,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeFieldsSerially(GraphQLType: Mutation, null, array(0), instance of ArrayObject(1))"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 208,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeOperation(instance of GraphQL\\Language\\AST\\OperationDefinitionNode, null)"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/Executor/Executor.php",
          "line": 155,
          "call": "GraphQL\\Executor\\ReferenceExecutor::doExecute()"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/GraphQL.php",
          "line": 158,
          "call": "GraphQL\\Executor\\Executor::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of GraphQL\\Type\\Schema, instance of GraphQL\\Language\\AST\\DocumentNode, null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null)"
        },
        {
          "file": "/var/www/vendor/webonyx/graphql-php/src/GraphQL.php",
          "line": 90,
          "call": "GraphQL\\GraphQL::promiseToExecute(instance of GraphQL\\Executor\\Promise\\Adapter\\SyncPromiseAdapter, instance of GraphQL\\Type\\Schema, 'mutation {\n  createMeal(input: {name: \"Mein Essen\", author: {connect: 1}}) {\n    id\n    name\n  }\n}', null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null, array(29))"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/GraphQL.php",
          "line": 156,
          "call": "GraphQL\\GraphQL::executeQuery(instance of GraphQL\\Type\\Schema, 'mutation {\n  createMeal(input: {name: \"Mein Essen\", author: {connect: 1}}) {\n    id\n    name\n  }\n}', null, instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null, array(29))"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/GraphQL.php",
          "line": 99,
          "call": "Nuwave\\Lighthouse\\GraphQL::executeQuery('mutation {\n  createMeal(input: {name: \"Mein Essen\", author: {connect: 1}}) {\n    id\n    name\n  }\n}', instance of Nuwave\\Lighthouse\\Schema\\Context, array(0), null, null)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Support/Http/Controllers/GraphQLController.php",
          "line": 59,
          "call": "Nuwave\\Lighthouse\\GraphQL::executeRequest(instance of Nuwave\\Lighthouse\\Execution\\LighthouseRequest)"
        },
        {
          "call": "Nuwave\\Lighthouse\\Support\\Http\\Controllers\\GraphQLController::query(instance of Nuwave\\Lighthouse\\Execution\\LighthouseRequest)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Controller.php",
          "line": 54,
          "function": "call_user_func_array(array(2), array(1))"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php",
          "line": 45,
          "call": "Illuminate\\Routing\\Controller::callAction('query', array(1))"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
          "line": 239,
          "call": "Illuminate\\Routing\\ControllerDispatcher::dispatch(instance of Illuminate\\Routing\\Route, instance of Nuwave\\Lighthouse\\Support\\Http\\Controllers\\GraphQLController, 'query')"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
          "line": 196,
          "call": "Illuminate\\Routing\\Route::runController()"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
          "line": 685,
          "call": "Illuminate\\Routing\\Route::run()"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 128,
          "call": "Illuminate\\Routing\\Router::Illuminate\\Routing\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Support/Http/Middleware/AttemptAuthentication.php",
          "line": 32,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Nuwave\\Lighthouse\\Support\\Http\\Middleware\\AttemptAuthentication::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/nuwave/lighthouse/src/Support/Http/Middleware/AcceptJson.php",
          "line": 29,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Nuwave\\Lighthouse\\Support\\Http\\Middleware\\AcceptJson::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 103,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
          "line": 687,
          "call": "Illuminate\\Pipeline\\Pipeline::then(instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
          "line": 662,
          "call": "Illuminate\\Routing\\Router::runRouteWithinStack(instance of Illuminate\\Routing\\Route, instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
          "line": 628,
          "call": "Illuminate\\Routing\\Router::runRoute(instance of Illuminate\\Http\\Request, instance of Illuminate\\Routing\\Route)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
          "line": 617,
          "call": "Illuminate\\Routing\\Router::dispatchToRoute(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
          "line": 165,
          "call": "Illuminate\\Routing\\Router::dispatch(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 128,
          "call": "Illuminate\\Foundation\\Http\\Kernel::Illuminate\\Foundation\\Http\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
          "line": 21,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php",
          "line": 21,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php",
          "line": 27,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php",
          "line": 63,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/fruitcake/laravel-cors/src/HandleCors.php",
          "line": 37,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Fruitcake\\Cors\\HandleCors::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/fideloper/proxy/src/TrustProxies.php",
          "line": 57,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 167,
          "call": "Fideloper\\Proxy\\TrustProxies::handle(instance of Illuminate\\Http\\Request, instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
          "line": 103,
          "call": "Illuminate\\Pipeline\\Pipeline::Illuminate\\Pipeline\\{closure}(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
          "line": 140,
          "call": "Illuminate\\Pipeline\\Pipeline::then(instance of Closure)"
        },
        {
          "file": "/var/www/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
          "line": 109,
          "call": "Illuminate\\Foundation\\Http\\Kernel::sendRequestThroughRouter(instance of Illuminate\\Http\\Request)"
        },
        {
          "file": "/var/www/public/index.php",
          "line": 55,
          "call": "Illuminate\\Foundation\\Http\\Kernel::handle(instance of Illuminate\\Http\\Request)"
        }
      ]
    }
  ],
  "data": {
    "createMeal": null
  }
}
```

</details></br>

**Lighthouse Version**
4.16

**Laravel Version**
7.0

**Comment by spawnia** [comment]
We have substantial tests that ensure `@belongsTo` works as expected, see https://github.com/nuwave/lighthouse/blob/master/tests/Integration/Execution/MutationExecutor/BelongsToTest.php

Maybe something is off with your table definition? Try connecting an author with plain PHP and see if that works. If you can reproduce and pinpoint an error within Lighthouse, I am happy to investigate further.

**Comment by theovier** [comment]
Thanks for the reply and for pointing me to the test cases @spawnia. Turns out, you are dead right - it was a mistake on my table definition. 

I named the database field `author`, exactly the same as the relationship method `author()` whereas I should have named the db filed `author_id`. Laravel then can't find any relationship `author() `because it keeps looking for the (wrongly) defined attribute with the same name.

TLDR; `$table->foreignId('author_id')->constrained('users');` fixes my issue. 

Sorry for any inconvenience caused.
